### PR TITLE
[#138279259] Blog post: dead code

### PR DESCRIPTION
**Pull request by runongirlrunon**
@pfarrell @beghbali 
[Tracker #138279259](https://www.pivotaltracker.com/story/show/138279259)

### CODE REVIEW

File: source/_posts/2017-01-24-burying-your-dead-code.markdown [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+author: "Lindsey Anne <lindsey.hogg@grandrounds.com>"
+categories: git bisect, dead code
+---
+As our code evolves and changes, little bits of leftovers from refactors or changed paths can build up as dead code. It's like the plaque that builds up on your teeth over time: not really noticeable at first, but after a trip to the dentist, you feel a whole lot better.
+
+Removing dead code from your codebase is a simple process at its core: find the dead parts, prove that they're dead, and delete them. The tricky part is in that second step, but after a little trial and error, we discovered a couple tricks to speed up the process.
+
+##### Start small
+
+Before digging into a gigantic repo to see what you can rip out of there, consider the impact of what you're about to do and start with just a small piece of it. Since I was investigating one of our biggest repos, I decided to tackle a single Model first rather than, say, the entire lib directory.
```
Comment (etripier) [inline source/_posts/2017-01-24-burying-your-dead-code.markdown]: I would say "Because I was investigating ...".

File: source/_posts/2017-01-24-burying-your-dead-code.markdown [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+##### Finding the dead parts
+
+I'm sure there's other gems out there that perform similar assessment, but our favorite gem for this is [debride by Seattle Ruby Brigade](https://github.com/seattlerb/debride). It's not 100% perfect, but it will give you a good starting place to start digging around. The whitelisting option is particularly helpful after your initial investigation. Another option is the brute-force method, where you comb through a file and investigate each and every piece of it one by one. If your `debride` output is suspiciously large, this might be worth a try anyway.
```
Comment (etripier) [inline source/_posts/2017-01-24-burying-your-dead-code.markdown]: "I'm sure there are other gems ..."

"it will give you a good starting place to start digging around" -> one of those "start"s should probably go.

"each and every piece of it one by one" -> vaguely tautological.

"this might be worth a try anyway" -> the "anyway" here is unnecessary.

File: source/_posts/2017-01-24-burying-your-dead-code.markdown [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+`git bisect` is generally used for tracking down when a bug was introduced by using a binary search through as many commits as you'd like; you can start at the very beginning of your git history or somewhere in the middle. It checks out the repo at each commit, where you can test your issue and mark it as "good" or "bad." Rather than testing for a bug, though, I used it to `grep` for the method names that were appearing as dead from my previous step one at a time. If `grep` only gave one result, then nothing else was calling the method anymore and I'd mark it "bad." If it showed up with two or more results (minus tests that might still exist), I'd mark it "good." The point when searching for a bug is to find the last "good" commit before a bug was introduced; the point when using it to look for dead code is to track down when the consumer was removed.
```
Comment (etripier) [inline source/_posts/2017-01-24-burying-your-dead-code.markdown]: I might include a sample block of code demonstrating `git bisect`. From your description it's not immediately apparent that you're searching through a range of commits using `git bisect good|bad ${rev}` - a visual example could be really helpful.

File: source/_posts/2017-01-24-burying-your-dead-code.markdown [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+##### Delete them
+
+Eventually, I'd come across the commit where the diff showed the changed or removed method call, and I could remove it in my current branch. After deleting all the dead code, I made sure to comment on each removed function in my pull request with the commit number where the consumer was removed (because documentation is rad). Lots of eyes and code review later, our Model was done with her teeth cleaning and felt just a little bit better.
```
Comment (etripier) [inline source/_posts/2017-01-24-burying-your-dead-code.markdown]: "Lots of eyes and code review later" -> I might rephrase this to remove the reference to "eyes" - although I know what you mean, it sounds a little disembodied. Also, "code reviews" should be plural to preserve parallel structure.

Comment (runongirlrunon) [inline source/_posts/2017-01-24-burying-your-dead-code.markdown]: How about `"Many code reviews later, ..."`?

Comment (etripier) [inline source/_posts/2017-01-24-burying-your-dead-code.markdown]: Yup, that works.

**Comment by runongirlrunon** [comment]
@etripier made your suggested edits and added an example `git bisect` search from jarvis. I scrubbed out all the specifics except for the SHAs--I think those are safe to leave in without giving away too many of our secrets.
